ruma-events/src/room/history_visibility.rs

```diff
@@ -10,9 +10,15 @@ use crate::StateEvent;
 /// from before they joined.
 pub type HistoryVisibilityEvent = StateEvent<HistoryVisibilityEventContent>;
 
+/// The redacted version of this event.
+///
+/// Since no keys are excluded post redaction we use a type alias to specify the redacted event.
+pub type RedactedHistoryVisibilityEventContent = HistoryVisibilityEventContent;
+
 /// The payload for `HistoryVisibilityEvent`.
 #[derive(Clone, Debug, Deserialize, Serialize, StateEventContent)]
 #[ruma_event(type = "m.room.history_visibility")]
+#[ruma_event(custom_redacted)]
```

Comment:
Why `custom_redacted`?

Reply:
These are just type aliases, this attribute skips the whole redacted code-gen.

Reply:
Oh, I see. I don't think we should have them as type aliases though. I don't think it buys us much, and having redaction be 'deserialize -> .redact() -> serialize' for most types but 'deserialize -> serialize' for just a few seems weird. Plus, we want it to be a backwards-compatible change to add optional fields that don't survive redaction (although I wonder how likely it is that this will actually occur).

Reply:
Ohh right it messes up all the methods, didn't think of that. I'll just remove that attribute for those events.